resourcet/Data/Acquire/Internal.hs

```diff
@@ -25,8 +27,21 @@ import qualified Control.Monad.Catch as C ()
 -- @since 1.1.2
 data ReleaseType = ReleaseEarly
                  | ReleaseNormal
-                 | ReleaseException
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+                 | ReleaseException' E.SomeException
+    deriving (Show, Typeable)
+
+-- | Fake 'E.Exception' to use with the deprecated 'ReleaseException' pattern.
+data DeprecatedReleaseExceptionPlaceholder = DeprecatedReleaseExceptionPlaceholder
+     deriving (Show)
+
+instance E.Exception DeprecatedReleaseExceptionPlaceholder
```

Comment:
If we keep construction, then we should probably attach a `DEPRECATION` pragma to this type too

Reply:
Why? The only reason a user would want to reference this is if they wanted to know that they're downstream of broken code, but in that case this isn't deprecated, there's no replacement.

Reply:
Suppose we complete the deprecation by deleting the `ReleaseException` pattern. Then there is no reason to have this around either, and it gets deleted. I'd call that 'deprecated'.

Reply:
Fake exception removed.